crates/syntax-bridge/src/lib.rs

```diff
@@ -982,7 +985,10 @@ where
                             format_to!(self.buf, "{lit}");
                             debug_assert_ne!(self.buf.len() - buf_l, 0);
                             self.text_pos += TextSize::new((self.buf.len() - buf_l) as u32);
-                            self.token_map.push(self.text_pos, lit.span);
+                            combined_span = match combined_span {
+                                None => Some(lit.span),
+                                Some(prev_span) => Some(Self::merge_spans(prev_span, lit.span)),
+                            };
```

Comment:
could do (I think the span is considered `Copy` here)
```suggestion
                            combined_span = combined_span.map_or(lit.span, |prev_span| Self::merge_spans(prev_span, lit.span));
```
but I leave that to you